pkg/client/query.go

```diff
@@ -101,53 +110,86 @@ func (c *QueryClient) Start() {
 }
 
 func (c *QueryClient) run() {
-	c.runQueryAndVerifyResult()
+	c.runQueries()
 
 	ticker := time.NewTicker(c.cfg.QueryInterval)
 
 	for {
 		select {
 		case <-ticker.C:
-			c.runQueryAndVerifyResult()
+			c.runQueries()
 		}
 	}
 }
 
-func (c *QueryClient) runQueryAndVerifyResult() {
+func (c *QueryClient) runQueries() {
 	// Compute the query start/end time.
 	start, end, ok := c.getQueryTimeRange(time.Now().UTC())
 	if !ok {
-		level.Debug(c.logger).Log("msg", "query skipped because of no eligible time range to query")
-		c.queriesTotal.WithLabelValues(querySkipped).Inc()
+		level.Debug(c.logger).Log("msg", "skipped querying because no eligible time range to query")
+		c.queriesTotal.WithLabelValues(querySkipped, "").Inc()
 		return
 	}
 
 	step := c.getQueryStep(start, end, c.cfg.ExpectedWriteInterval)
 
-	samples, err := c.runQuery(start, end, step)
+	wg := sync.WaitGroup{}
+	wg.Add(1 + len(c.cfg.AdditionalQueries))
+
+	go func() {
+		defer wg.Done()
+
+		c.runDefaultQuery(start, end, step)
+	}()
+
+	for _, query := range c.cfg.AdditionalQueries {
+		go func() {
+			defer wg.Done()
+
+			c.runAdditionalQuery(start, end, step, query)
+		}()
+	}
```

Comment:
Any reason to run queries in parallel here?

Reply:
My thinking was that it would useful when you're trying to stress a system, but I'm not wedded to this, happy to make it sequential.